fastlane/actions/get_latest_github_release_within_same_major_action.rb

```diff
@@ -0,0 +1,99 @@
+module Fastlane
+  module Actions
+
+    class GetLatestGithubReleaseWithinSameMajorAction < Action
+      def self.run(params)
+        repo_name = params[:repo_name]
+        UI.message("Getting latest release for #{repo_name} on GitHub within same major as the current version")
+
+        GithubApiAction.run(
+          server_url: "https://api.github.com",
+          api_token: ENV["GITHUB_API_TOKEN"],
+          api_bearer: nil,
+          http_method: 'GET',
+          path: "repos/RevenueCat/#{repo_name}/releases",
+          error_handlers: {
+            404 => proc do |result|
+              UI.error("Repository #{repo_name} cannot be found, please double check its name and that you provided a valid API token (if it's a private repository).")
+            end,
+            401 => proc do |result|
+              UI.error("You are not authorized to access #{repo_name}, please make sure you provided a valid API token.")
+            end,
+            '*' => proc do |result|
+              UI.error("GitHub responded with #{result[:status]}:#{result[:body]}")
```

Comment:
Hmm I wonder if these should be `UI.user_error!` instead... I think with these, the lane will continue even if the request fails...

Reply:
Hmm, maybe yes. I got inspired by this official action https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/get_github_release.rb#L19

And they do `error` there. But happy to change it to `user_error`.

Reply:
Ok, yes I am going to change them. Looking at fastlane docs for `user_error`:

```
 fastlane "crash" because of a user error everything that is caused by the user and is not unexpected
 ```
 
 It looks like most of these errors are caused by a bad configuration, and we want to crash, so I will change them.